server/src/main/java/org/opensearch/search/startree/filter/provider/DimensionFilterMapper.java

```diff
@@ -88,6 +92,20 @@ Optional<Long> getMatchingOrdinal(
         DimensionFilter.MatchType matchType
     );
 
+    /**
+     * Returns the dimensionDataType used for comparing dimension values. <br>
+     * This determines how numeric values are compared: <br>
+     *   - DimensionDataType.UNSIGNED_LONG for unsigned long values <br>
+     *   - DimensionDataType.LONG for all other numeric types (DEFAULT)
+     */
+    default DimensionDataType getDimensionDataType() {
+        return DimensionDataType.LONG;
+    }
+
+    default Comparator<Long> comparator() {
```

Comment:
We can rename to explicitly show that this is for comparing doc values. Maybe `docValueComparator`

Reply:
This is not just for doc values - used for both star node and dv comparison both.